microprofile/grpc/client/src/main/java/io/helidon/microprofile/grpc/client/GrpcProxyProducer.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.grpc.client;
+
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.InjectionPoint;
+
+import io.helidon.grpc.client.GrpcChannelsProvider;
+import io.helidon.microprofile.grpc.core.InProcessGrpcChannel;
+import io.helidon.microprofile.grpc.core.model.ModelHelper;
+
+import io.grpc.Channel;
+
+/**
+ * A utility class of gRPC CDI producer stubs.
+ * <p>
+ * The methods in this class are not real CDI producer methods,
+ * they act as templates that the {@link GrpcClientCdiExtension}
+ * will use to create producers on the fly as injection points
+ * are observed.
+ */
+class GrpcProxyProducer {
+
+    private GrpcProxyProducer() {
+    }
+
+    /**
+     * A CDI producer method that produces a client proxy for a gRPC service that
+     * will connect to the server using the default {@link Channel}.
+     * <p>
+     * This is not a real producer method but is used as a stub by the gRPC client
+     * CDI extension to create real producers as injection points are discovered.
+     *
+     * @param injectionPoint the injection point where the client proxy is to be injected
+     * @return a gRPC client proxy
+     */
+    @GrpcServiceProxy
+    public static Object proxyUsingDefaultChannel(InjectionPoint injectionPoint, ChannelProducer producer) {
```

Comment:
Class is package private, methods `public` - is this needed to correctly process the `GrpcServiceProxy` annotation, or can we decrease the visibility?